### PR TITLE
Upgrade to Gradle 8 and use kover for code coverage

### PR DESCRIPTION
**Pull request by popematt**
**Issue #, if available:**

#221 

**Description of changes:**

Removes `jacoco` and adds `kover`. Had to upgrade Gradle to get it to work with Kover, so I upgraded from 6.x to 8.x. As a result, there are some minor changes that are seemingly unrelated. Basically, if "jacoco" is not in a particular part of the diff, then it's a change for upgrading to Gradle 8.

**Related PRs in ion-schema, ion-schema-tests, ion-schema-schemas:**

None

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: gradlew
Comment (popematt) [inline gradlew:1]: 🗺️ This file was modified automatically by the Gradle wrapper upgrade tool.

File: gradlew.bat
Comment (popematt) [inline gradlew.bat:1]: 🗺️ This file was modified automatically by the Gradle wrapper upgrade tool.

File: gradle/wrapper/gradle-wrapper.properties
Comment (popematt) [inline gradle/wrapper/gradle-wrapper.properties:1]: 🗺️ This file was modified automatically by the Gradle wrapper upgrade tool.

File: cli/build.gradle [hunk 1]
```diff
@@ -38,7 +38,7 @@ dependencies {
 }
 
 application {
-    mainClassName = "com.amazon.ionschema.cli.MainKt"
+    mainClass.set("com.amazon.ionschema.cli.MainKt")
```
Comment (popematt) [inline cli/build.gradle:41]: 🗺️ `mainClassName` is removed in Gradle 8.

File: gradle/wrapper/gradle-wrapper.jar
Comment (popematt) [inline gradle/wrapper/gradle-wrapper.jar:1]: 🗺️ This file was modified automatically by the Gradle wrapper upgrade tool.

File: ion-schema/build.gradle [hunk 2]
```diff
@@ -65,12 +65,13 @@ dokkaJavadoc.configure {
 }
 
 task sourcesJar(type: Jar) {
-  classifier 'sources'
+  archiveClassifier.set("sources")
```
Comment (popematt) [inline ion-schema/build.gradle:68]: 🗺️  `classifier` was also removed in Gradle 8.

**Review by popematt — COMMENTED** [review]
To upgrade Gradle, I ran `gradle wrapper --gradle-version 8.1.1`. I've noted which changes were done by the tool rather than by me so that you don't waste your time looking at them.

**Review by jobarr-amzn — APPROVED** [review]

**Pull request merged**